Scripts/Tank.gd

```diff
@@ -4,34 +4,41 @@ extends KinematicBody2D
 export var speed :int = 200 
 
 # Speed to rotate the tank to the left/right
-export var rotation_speed :int = 200
+export var rotation_speed :int = 1.5
 
 # Delay in seconds between shoots
 export var shoot_delay :float = 1
 var current_shoot_delay :float = 0
 
+# "Private" variable to move the tank
+var velocity :Vector2 = Vector2.ZERO
+
 func _process(delta :float):
 	# Update timers	
 	current_shoot_delay = clamp(current_shoot_delay + delta, 0, shoot_delay)
 	
 	# Shoot
-	if Input.is_action_pressed("shoot") and current_shoot_delay == shoot_delay:
+	if Input.is_action_pressed("p1_shoot") and current_shoot_delay == shoot_delay:
 		print("Shoot") # TODO: Implement this action
 		current_shoot_delay = 0
 
 func _physics_process(delta :float):
-	# Update rotation
-	if Input.is_action_pressed("rotate_left"):
-		rotate(deg2rad(-rotation_speed) * delta)
-	if Input.is_action_pressed("rotate_right"):
-		rotate(deg2rad(rotation_speed) * delta)
-	
-	# Calculate forward vector
-	var current_rotation := transform.get_rotation()
-	var forward_vector := Vector2.UP.rotated(current_rotation)
+	# Rotate
+	var rotation_direction = _input_rotate()
+	rotation += rotation_direction * rotation_speed * delta
 	
-	# Update movement
-	if Input.is_action_pressed("move_forward"):
-		move_and_collide(forward_vector * rotation_speed * delta)
-	if Input.is_action_pressed("move_backwards"):
-		move_and_collide(-forward_vector * rotation_speed * delta)
+	# Move
+	_input_move()
+	velocity = velocity.rotated(rotation)
+	velocity = move_and_slide(velocity)
+	pass
+
+## Update rotation
+## This function must be override
+func _input_rotate() -> float:
```

Comment:
Probably in the future, we will want to add bots and something like that, and this will need a refactoring, but I think for now it's OK? 😅

Reply:
Humm, ye, for now it's just the name of the function,but yee, nothing to worry